src/sql/platform/connection/common/utils.ts

```diff
@@ -110,12 +110,19 @@ export function getUriPrefix(ownerUri: string): string {
 }
 
 export function generateUriWithPrefix(connection: IConnectionProfile, prefix: string): string {
-	let id = connection.getOptionsKey();
+	let id = connection.getOptionsKey() + '|id:' + connection.id;
```

Comment:
Before you defaulted to connection.serverName + ':' + connection.databaseName if connection.getOptionsKey() was undefined, but now you'll just end up with something like undefined|id:12345. Assuming you want to keep that behavior something like this would be better :

```
return prefix +  (connection.getOptionsKey() ||  connection.serverName + ':' + connection.databaseName) + '|id:' + connection.id;
```

and then